src/subscriptions/ConnectionSub.re

```diff
@@ -0,0 +1,72 @@
+type state_t =
+  | Uninitialized
+  | Init
+  | TryOpen
+  | Open
+  | Closed;
+
+type channel_t = {
+  port: string,
+  counterpartyPort: string,
+  channelID: string,
+  counterpartyChannelID: string,
+  state: state_t,
+  order: string,
+};
+
+type t = {
+  clientID: string,
+  connectionID: string,
+  counterpartyChainID: string,
+  counterpartyClientID: string,
+  counterpartyConnectionID: string,
+  channels: array(channel_t),
+};
+
+exception NotMatch(string);
+let parseState =
+  fun
+  | 0 => Uninitialized
+  | 1 => Init
+  | 2 => TryOpen
+  | 3 => Open
+  | 4 => Closed
+  | x => raise(NotMatch({j|This $x doesn't match any state.|j}));
+
+let parseOrder =
+  fun
+  | "0" => "None"
+  | "1" => "Unordered"
+  | "2" => "Ordered"
+  | x => raise(NotMatch({j|This $x doesn't match any state.|j}));
+
+module MultiConfig = [%graphql
+  {|
+  subscription Blocks($chainID: String!) {
+    connections(where: {counterparty_chain: {chain_id: {_eq: $chainID}}}) @bsRecord {
```

Comment:
I think we can wire up the search function to this,

another one is to change _eq to _ilike